src/server/qgsserverinterface.h

```diff
@@ -70,13 +70,6 @@ class SERVER_EXPORT QgsServerInterface
      */
     virtual void setRequestHandler( QgsRequestHandler *requestHandler ) = 0 SIP_SKIP;
```

Comment:
Does this need a note in the api docs about ownership?

Reply:
Yes, thanks.

btw I'm not so sure I'll merge this, given the not so enthusiastic comments in the ML and here.

Long story: I never said that this small PR would make the whole server thread safe, I know that there are a lot of other classes and computing paths that need to be checked and made thread safe.

I also know that multiprocessing is a viable option for scaling and I don't need to stress the differences from multi-threading and multi-processing in server applications, we all know the advantages and disadvantages of both solutions (that are not mutually exclusive!).

 This PR comes from a series of multi-threading tests that we were executing in the QGIS desktop side and we needed a *fast* and configurable XYZ server, that's why I adapted the server to be multi-threaded in  https://github.com/qgis/QGIS/commit/f29c9c99656b219209efec66b46818c06ff7433c and that's why I realized that the server plugin filters were occasionally processing wrong requests.

No problems to close this PR of course, but we should probably decide if we want to see a threaded server in the future, because if the answer is positive this PR would probably be a first micro-step in the right direction.